### PR TITLE
Internal: Use sharding for splitting tests

### PR DESCRIPTION
**Pull request by yotamselementor**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: .github/workflows/playwright.yml [hunk 1]
```diff
@@ -105,18 +90,16 @@ jobs:
         run: |
           npx playwright install chromium
       - name: Run Playwright tests
-        if: ${{matrix.testSuite !='elements-regression'}}
-        env:
-          TEST_SUITE: "@${{matrix.testSuite}}"
-        run: npm run test:playwright -- --reporter=${{inputs.reporter || 'list,github'}}
+        if: ${{ matrix.shardIndex !='elements-regression' }}
```
Comment (davseve) [inline .github/workflows/playwright.yml:93]:
```suggestion
        if: ${{ matrix.shardIndex != 'elements-regression' }}
```

File: .github/workflows/playwright.yml [hunk 2]
```diff
@@ -41,32 +41,17 @@ jobs:
...
-          'pluginTester2_containers',
-          'pluginTester1_sections',
-          'pluginTester2_sections',
-          'promotions',
-          'nested-atomic-repeaters',
-        ]
+        shardIndex: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
+        shardTotal: [ 10 ]
+        include:
+        - shardIndex: "elements-regression"
```
Comment (davseve) [inline .github/workflows/playwright.yml:54]: Are these lines needed? 
Aren't  `elements-regression` tests added by default like all the other tests?

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -105,18 +90,16 @@ jobs:
...
-        if: ${{matrix.testSuite !='elements-regression'}}
-        env:
-          TEST_SUITE: "@${{matrix.testSuite}}"
-        run: npm run test:playwright -- --reporter=${{inputs.reporter || 'list,github'}}
+        if: ${{ matrix.shardIndex !='elements-regression' }}
+        run: npm run test:playwright  -- --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }} --reporter=${{ inputs.reporter || 'list,github' }}
       - name: Run element regression tests
-        if: ${{matrix.testSuite =='elements-regression'}}
-        run: npm run test:playwright:elements-regression -- --reporter=${{inputs.reporter || 'list,github'}}
+        if: ${{ matrix.shardIndex =='elements-regression' }}
```
Comment (davseve) [inline .github/workflows/playwright.yml:96]:
```suggestion
        if: ${{ matrix.shardIndex == 'elements-regression' }}
```

File: tests/playwright/assets/breakpoints.ts [hunk 4]
```diff
@@ -44,4 +44,18 @@ export default class {
...
 		await this.page.waitForSelector( '#elementor-editor-wrapper' );
 	}
+
+	async resetBreakpoints() {
+		await this.page.click( '#elementor-panel-footer-responsive' );
+		await this.page.click( '#e-responsive-bar__settings-button' );
+		await this.page.waitForSelector( 'text=Active Breakpoints' );
+
+		const removeBreakpointButton = '#elementor-kit-panel-content .select2-selection__choice__remove';
+		while ( await this.page.$( removeBreakpointButton ) ) {
```
Comment (davseve) [inline tests/playwright/assets/breakpoints.ts:54]: `$` is [deprecated](elementor-panel-footer-responsive)
```suggestion
		while ( await this.page.locator( removeBreakpointButton ) ) {
```

Comment (yotamselementor) [reply tests/playwright/assets/breakpoints.ts]: Fixed in https://github.com/elementor/elementor/pull/27617/commits/ba404aa8d49a71f63e0279353aac83b8961b6182

File: .github/workflows/playwright.yml (hunk 1 above)
Comment (yotamselementor) [reply .github/workflows/playwright.yml]: Fixed in https://github.com/elementor/elementor/pull/27617/commits/ba404aa8d49a71f63e0279353aac83b8961b6182

File: .github/workflows/playwright.yml (hunk 2 above)
Comment (yotamselementor) [reply .github/workflows/playwright.yml:54]: No - this means that on top of the shards, we have another scenario with this value

**Review by davseve — APPROVED** [review]

**Pull request merged**